### ISSUE TITLE
cz vs z check and fixing by speed of light on comoving distances

### ISSUE THREAD

**Issue by misharash** [manodeep/Corrfunc#274]
## General information

- Corrfunc version: 2.4.0
- platform: GNU/Linux
- installation method: pip

but I'm pretty sure the issue is general.

## Issue description
I am using `Corrfunc.mocks.DDsmu_mocks.DDsmu_mocks` in python with `is_comoving_dist=True`.

### Expected behavior 
No cz vs z check and no multiplication by speed of light when `is_comoving_dist=True`.

### Actual behavior
I am getting warnings like
```
check_ra_dec_cz_s_mu_double> Out of range values found for cz. Expected input to be `cz' but found `z' instead. max_cz (found in input) = 4.190300 threshold = 10.000000
```
I wondered how the code doesn't fail, traced it [down to this message](https://github.com/manodeep/Corrfunc/blob/68973d0681276a70e43dc882f15c514d222e24d1/mocks/DDsmu_mocks/countpairs_s_mu_mocks_impl.c.src#L86) and found that then [multiplication by speed of light occurs to "fix" the data](https://github.com/manodeep/Corrfunc/blob/68973d0681276a70e43dc882f15c514d222e24d1/mocks/DDsmu_mocks/countpairs_s_mu_mocks_impl.c.src#L98), which doesn't make sense with distances.

## P.S.
With distances in Mpc/h, such small maximum values are also a problem for me, but the fixing procedure for cz/z is not applicable, and the threshold of 10 might not be the best too.

**Comment by manodeep** [comment]
Good catch! Wonder if this is related to #273?

A similar "feature" should also exist for `DDrppi_mocks` - do you mind applying the fix there as well?

**Comment by misharash** [comment]
Sure. Indeed `DDrppi_mocks` was pretty much the same. Correction for that is now also included in #275.

#273 doesn't seem related - distance values in example there are large so correction shouldn't be invoked. This thing I noticed is very particular and probably won't make real difference for any correct data, mine just happened to be messed up.
